chainio/clients/elcontracts/reader.go

```diff
@@ -60,29 +67,63 @@ type ELReader interface {
 }
 
 type ELChainReader struct {
-	logger            logging.Logger
-	elContractsClient clients.ELContractsClient
-	ethClient         eth.EthClient
+	logger                  logging.Logger
+	slasher                 slasher.ContractSlasherCalls
+	delegationManager       delegationmanager.ContractDelegationManagerCalls
+	strategyManager         strategymanager.ContractStrategyManagerCalls
+	blsPubkeyCompendium     blspkcompendium.ContractBLSPublicKeyCompendiumCalls
+	blsPubKeyCompendiumAddr common.Address
+	ethClient               eth.EthClient
 }
 
 // forces EthReader to implement the chainio.Reader interface
 var _ ELReader = (*ELChainReader)(nil)
 
 func NewELChainReader(
```

Comment:
Do we want to make this private if we want to enforce them to use `BuildELChainReader`?

Reply:
We discussed offline. Let's keep it for now. It's used in the chainio/cleints/builder constructor. We can revisit the design of these clients (what we want private, what we want public, which constructors are required, which should we hide, etc.) in a later refactor.